### ISSUE TITLE
ENOENT: no such file or directory, open '/tmp/automation-logs-preview-WtufVl/eventlog.txt'

### ISSUE THREAD

**Issue by jccr** [pulumi/actions#163]
I need some help with Pulumi and GitHub actions.
So far it's been working fine until recently I've been getting this error:
```
Run pulumi/actions@v2
  with:
    command: preview
    stack-name: ***
    work-dir: ./
    comment-on-pr: false
    parallel: 2147483647
    target-dependents: false
    refresh: false
    upsert: false
  env:
    NODE_ENV: production
    PULUMI_ACCESS_TOKEN: ***
    PULUMI_STACK_NAME: ***
    npm_config_yes: true
pulumi preview on ***
  Error: ENOENT: no such file or directory, open '/tmp/automation-logs-preview-WtufVl/eventlog.txt'
```
## Expected behavior
Pulumi Preview succeeds.

## Current behavior
Preview succeeds on the Pulumi Console side, but this error interrupts the actions workflow.

## Steps to reproduce

1. Create a workflow with preview action
2. Run the workflow

## Context (Environment)
I'm trying to run a preview action on a pull_request event.
I have the Pulumi bot running as well, and that's working fine and reporting that the preview succeeded.

**Comment by cobraz** [comment]
Thank you for posting this issue.

I have a few questions that would help a lot if you could answer.

- Which OS are you running on? `runs-on`
- Does this error happen at every run?
- Are you using managed or self-hosted Github Action?

**Comment by jccr** [comment]
@cobraz 
> I have a few questions that would help a lot if you could answer.
> 
> * Which OS are you running on? `runs-on`

Using `ubuntu-latest`

> * Does this error happen at every run?

It appears so, yes.

> * Are you using managed or self-hosted Github Action?

Managed by GitHub.

> If you don't mind, I would love to take a look at your workflow file.

For sure, here it is:

```yml
name: Pulumi

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

env:
  NODE_ENV: production
  PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
  PULUMI_STACK_NAME: redacted-user/redacted-org/dev
  npm_config_yes: true

jobs:
  pulumi:
    runs-on: ubuntu-latest
    steps:
      # Turnstyle is used to prevent multiple push jobs from running at the same time. We
      # limit it to push jobs to allow PR jobs to run concurrently.
      - name: Turnstyle
        if: github.event_name == 'push'
        uses: softprops/turnstyle@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

      - uses: actions/checkout@v2

      - name: Use Node.js
        uses: actions/setup-node@v1
        with:
          node-version: 14.x

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo "::set-output name=dir::$(yarn cache dir)"

      - uses: actions/cache@v2
        id: yarn-cache
        with:
          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            ${{ runner.os }}-yarn-

      - name: Install Pulumi CLI
        uses: pulumi/action-install-pulumi-cli@v1.0.1

      - name: Restore npm dependencies
        run: yarn install --prod

      - name: PR previews
        if: github.event_name == 'pull_request'
        uses: pulumi/actions@v2
        with:
          command: preview
          stack-name: ${{ env.PULUMI_STACK_NAME }}

      - name: Apply infrastructure update
        if: github.event_name == 'push'
        uses: pulumi/actions@v2
        with:
          command: up
          stack-name: ${{ env.PULUMI_STACK_NAME }}
```

**Comment by jccr** [comment]
Follow up from my last comment,

I've switched from this:

```yml
- name: PR previews
  if: github.event_name == 'pull_request'
  uses: pulumi/actions@v2
  with:
    command: preview
    stack-name: ${{ env.PULUMI_STACK_NAME }}
```

To this:
```yml
- name: PR previews
  if: github.event_name == 'pull_request'
  run: pulumi preview -s $PULUMI_STACK_NAME
```

... and with this workaround I am seeing my preview succeed without errors, as expected. (I've changed nothing else but this.)

**Comment by EvanBoyle** [comment]
@komalali is there a potential race condition? Looks like we might fail if the file doesn't yet exist

**Comment by komalali** [comment]
I've chatted with both @cobraz and @stack72 about this. My suspicion is that there is some issue with creating tempfiles in github actions, or that this error is masking a different issue. 

We [wait for 10 seconds](https://github.com/pulumi/pulumi/blob/66f5d50a5039f58d92042c76dcf66517f600b9fb/sdk/nodejs/x/automation/stack.ts#L135-L138) for the log file to be created, after which we fail with a file not found error. In the preview case we depend on the event log to get the [changeSummary](https://github.com/pulumi/pulumi/blob/66f5d50a5039f58d92042c76dcf66517f600b9fb/sdk/nodejs/x/automation/stack.ts#L349), so not sure we can "opt out".

It's unclear why this has just started popping up in the last few days, as this change was released a few weeks ago, and it's not as widespread as I would expect if there was a GHA issue.

**Comment by EvanBoyle** [comment]
We should upgrade the version of the nodejs SDK used in the action once https://github.com/pulumi/pulumi/pull/6730 is release and see if that resolves the issue.

**Comment by LucasRejanio** [comment]
I have the same problem, the most interesting is that it occurs in only one specific stack, in the others I use it is working normally. It is really a very peculiar mistake!

**Comment by komalali** [comment]
v2.25.0 is out now and includes @EvanBoyle's fix so we should give that a go and see if that helps with the issues we're seeing here.

**Comment by stack72** [comment]
Hi @jccr and @LucasRejanio 

I just published a release that depends on v2.25.0 as @komalali said above - we believe this should fix the issue

PLEASE NOTE: The CLI requirement for this fix is v2.25.0 

Please let me know if this is not fixed and I will reopen the issue

Paul

**Comment by LucasRejanio** [comment]
Execelente @stack72  I did some tests and it worked perfectly with this new release. Thank you Pulumi team for the solution!
